File: LeagueItemExplorerFront/src/app/models/item-data.ts
import { ItemDataGold } from "./item-data-gold";
import { ItemDataImage } from "./item-data-image";

export interface ItemData {
    name: string;
    description: string;
    colloq: string;
    plaintext: string;
    into: string[];
    tags: string[];
    image : ItemDataImage;
    gold: ItemDataGold;
    maps: { [key: string]: boolean };
    stats: { [key: string]: number };
}

File: LeagueItemExplorerJAR/LeagueItemExplorerJAR/src/main/java/com/app/LeagueItemExplorerJAR/models/ItemData.java
package com.app.LeagueItemExplorerJAR.models;

import lombok.Data;

import java.util.Map;

@Data
public class ItemData {
    private String name;
    private String description;
    private String colloq;
    private String plaintext;
    private String[] into;
    private ItemDataImage image;
    private ItemDataGold gold;
    private String[] tags;
    private Map<String,Boolean> maps;
    private Map<String,Integer> stats;

}
